agimus_controller/agimus_controller/warm_start_reference.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+import numpy.typing as npt
+from typing_extensions import override
+
+import pinocchio as pin
+
+from agimus_controller.warm_start_base import WarmStartBase
+from agimus_controller.trajectory import TrajectoryPoint
+
+
+class WarmStartReference(WarmStartBase):
+    """
+    A class for generating warmstart values for trajectory optimization problem.
+
+    This class uses a reference trajectory and the robot model to compute the initial state,
+    state vectors, and control inputs.
+
+    Attributes:
+    _rmodel (pin.Model | None): The robot's Pinocchio model, used for forward dynamics computations.
+    _rdata (pin.Data | None): Data structure associated with the Pinocchio model.
+
+    Methods:
```

Comment:
As mentioned before. This documentation will be inherited from the base class and method docstrings itself